File: src/Asm.MooBank/Models/AccountHolder.cs
namespace Asm.MooBank.Models;

public partial record AccountHolder
{
    public Guid Id { get; set; }

    public string? EmailAddress { get; set; }

    public string? FirstName { get; set; }

    public string? LastName { get; set; }

    public Guid FamilyId { get; set; }

    public Guid? PrimaryAccountId { get; set; }

    public IEnumerable<Guid> Accounts { get; set; } = Enumerable.Empty<Guid>();
}

File: src/Asm.MooBank.Web.App/src/models/AccountHolder.ts
export interface AccountHolder {
    id: string,
    emailAddress : string,
    firstName?: string,
    lastName?: string,
    currency: string
    familyId: string,
    primaryAccountId: string,
    cards: Card[]
}

export interface Card {
    last4Digits: number,
    name: string
}
